geoconvert/convert.py

```diff
@@ -413,8 +414,15 @@ def _full_name_to_country_code(text, lang, language_to_full_names):
         # If the language is unknown, just do not use any language.
         language_to_full_names = {}
 
-    for lang, full_names in language_to_full_names.items():
-        country_code = _full_name_to_country_code_for_lang(text, lang, full_names)
+    for language, full_names in language_to_full_names.items():
+        # When language is not set, we remove ambiguous countries names from the countries list
+        if not lang:
+            full_names = {
+                name: code
+                for name, code in full_names.items()
+                if name not in ambiguous_countries
```

Comment:
Wouldn't it be more efficient to loop over the ambiguous country names, and delete them from the (deep copy of ?) dict ?

Reply:
That's what I did at first but thought it made the code uglier, I'll change it back. No need for a deep copy though, it's a simple dict[str, str].